File: core/src/main/java/ru/surfstudio/android/core/app/SharedPrefModule.java
package ru.surfstudio.android.core.app;

import android.content.Context;
import android.content.SharedPreferences;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;
import ru.surfstudio.android.dagger.scope.PerApplication;

@Module
public class SharedPrefModule {

    public static final String NO_BACKUP_SHARED_PREF = "NO_BACKUP_SHARED_PREF";
    public static final String BACKUP_SHARED_PREF = "BACKUP_SHARED_PREF";

    @Provides
    @PerApplication
    @Named(SharedPrefModule.NO_BACKUP_SHARED_PREF)
    public SharedPreferences provideNoBackupSharedPref(Context context){
        return context.getSharedPreferences(NO_BACKUP_SHARED_PREF, Context.MODE_PRIVATE);
    }

    @Provides
    @PerApplication
    @Named(SharedPrefModule.BACKUP_SHARED_PREF)
    public SharedPreferences provideBackupSharedPref(Context context){
        return context.getSharedPreferences(BACKUP_SHARED_PREF, Context.MODE_PRIVATE);
    }
}

File: template/base_feature/src/main/java/ru/surfstudio/standard/application/storage/di/SharedPrefModule.kt
package ru.surfstudio.standard.application.storage.di

import android.content.Context
import android.content.SharedPreferences
import dagger.Module
import dagger.Provides
import ru.surfstudio.android.dagger.scope.PerApplication
import ru.surfstudio.android.shared.pref.BACKUP_SHARED_PREF
import ru.surfstudio.android.shared.pref.NO_BACKUP_SHARED_PREF
import javax.inject.Named

@Module
class SharedPrefModule {

    @Provides
    @PerApplication
    @Named(NO_BACKUP_SHARED_PREF)
    fun provideNoBackupSharedPref(context: Context): SharedPreferences {
        return context.getSharedPreferences(NO_BACKUP_SHARED_PREF, Context.MODE_PRIVATE)
    }

    @Provides
    @PerApplication
    @Named(BACKUP_SHARED_PREF)
    fun provideBackupSharedPref(context: Context): SharedPreferences {
        return context.getSharedPreferences(BACKUP_SHARED_PREF, Context.MODE_PRIVATE)
    }
}